lib/src/git_backend.rs

```diff
@@ -391,6 +392,48 @@ impl GitBackend {
             .map_err(|err| to_read_object_err(err, id))?;
         Ok(Box::new(Cursor::new(blob.take_data())))
     }
+
+    fn new_diff_platform(&self) -> BackendResult<gix::diff::blob::Platform> {
+        let attributes = gix::worktree::Stack::new(
+            self.git_repo().path(),
+            gix::worktree::stack::State::AttributesStack(Default::default()),
+            gix::worktree::glob::pattern::Case::Sensitive,
+            Vec::new(),
+            Vec::new(),
+        );
+        let filter = gix::diff::blob::Pipeline::new(
+            Default::default(),
+            gix_filter::Pipeline::new(
+                self.git_repo()
+                    .command_context()
+                    .map_err(|err| BackendError::Other(Box::new(err)))?,
+                Default::default(),
+            ),
+            Vec::new(),
+            Default::default(),
+        );
+        Ok(gix::diff::blob::Platform::new(
+            Default::default(),
+            filter,
+            gix::diff::blob::pipeline::Mode::ToGit,
+            attributes,
+        ))
+    }
+
+    async fn read_tree_for_commit<'repo>(
+        &self,
+        repo: &'repo gix::Repository,
+        id: &CommitId,
+    ) -> BackendResult<gix::Tree<'repo>> {
+        let tree = self.read_commit(id).await?.root_tree.to_merge();
```

Comment:
I made `read_tree_for_commit` sync and called `block_on` in its implementation instead of pushing that to its caller.  I thought about pulling out a `read_commit_sync()` but `read_commit().block_on()` seems about the same in terms of cognitive overhead and functions identially.